### PR TITLE
Migrate i3-nagbar to draw_util

### PR DESCRIPTION
**Pull request by mihaicmn**
See #2642 for details.

### CODE REVIEW

File: i3-nagbar/main.c [hunk 1]
```diff
@@ -191,107 +197,63 @@ static void handle_button_release(xcb_connection_t *conn, xcb_button_release_eve
...
 /*
+ * Draws a button and returns its width
+ *
+ */
+static int button_draw(button_t *button, int position) {
+    int text_width = predict_text_width(button->label);
+    button->width = text_width + 2 * BTN_PADDING + 2 * BTN_BORDER;
+    button->x = position - button->width;
+
+    /* draw border (TODO implement draw_util_stroke) */
```
Comment (Airblader) [inline i3-nagbar/main.c]: Open TODO

File: i3-nagbar/main.c [hunk 2]
```diff
@@ -191,107 +197,63 @@ static void handle_button_release(xcb_connection_t *conn, xcb_button_release_eve
...
-        set_font_colors(pixmap_gc, color_text, color_button_background);
-        /* the x term seems to set left/right padding */
-        draw_text(buttons[c].label, pixmap, pixmap_gc, NULL,
-                  y - w - line_width + logical_px(6),
-                  logical_px(4) + logical_px(3),
-                  rect.width - y + w + line_width - logical_px(6));
-
-        y -= w;
+    for (int i = 0; i < buttoncnt; i++) {
+        position -= logical_px(20);
```
Comment (Airblader) [inline i3-nagbar/main.c]: Isn't this something we should express in terms of extracted variables?

File: libi3/draw_util.c [hunk 3]
```diff
@@ -163,6 +163,33 @@ void draw_util_rectangle(surface_t *surface, color_t color, double x, double y,
 }
 
 /**
+ * Draws a line from (x1, y1) to (x2, y2).
+ *
+ */
+void draw_util_line(surface_t *surface, color_t color, double width, int x1, int y1, int x2, int y2) {
```
Comment (Airblader) [inline libi3/draw_util.c]: There's actually a reason we didn't add this function. It actually doesn't quite work as we want it to, see https://github.com/i3/i3/pull/1933#issuecomment-145444901

File: i3-nagbar/main.c (hunk 1 above)
Comment (mihaicmn) [inline i3-nagbar/main.c]: No longer needed due to the unwanted behavior of _cairo_stroke_ function.

**Comment by mihaicmn** [comment]
@Airblader thank you for the feadback. I updated the PR based on your comments.

**Comment by stapelberg** [comment]
@Airblader What’s your verdict? Is this good to go?

**Pull request merged**